src/relay/backend/utils.cc

```diff
@@ -227,6 +227,13 @@ Array<Pass> GetPassPrefix(bool is_homegeneous, bool is_vm) {
     pass_seqs.push_back(transform::EtaExpand(
         /* expand_constructor */ true, /* expand_global_var */ false));
   } else {
+    // DynamicToStatic runs FoldConstant, which affects SimplifyExpr below.
+    // Task extraction uses the is_vm=true branch, meaning SimplifyExpr sees different
+    // inputs from the ones when invoked via relay.build(...).
+    // This causes workload lookups in ApplyHistoryBest to fail if the lookup depends on
+    // the structual hash of the input relay module (e.g. MetaScheduler).
+    // TODO(masahi): Either remove DynamicToStatic below or always run it
```

Comment:
cc @junrushao1994 on this comment. I found this issue when working on applying my TIR VNNI conv2d to quantized resnet. Currently the workload lookup at https://github.com/apache/tvm/blob/ce335c3a74185df6cc1152e53c60695d8a418d8e/src/meta_schedule/integration.cc#L134 completely fails because of this, so it needs to be fixed.

I'll open an issue and take action next week.